nifi-nar-bundles/nifi-standard-services/nifi-standard-services-api-nar/pom.xml

```diff
@@ -26,6 +26,12 @@
         <source.skip>true</source.skip>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-jetty-bundle</artifactId>
+            <version>1.9.0-SNAPSHOT</version>
+            <type>nar</type>
```

Comment:
I'm curious what caused this to be needed. At runtime we automatically make the Jetty bundle be the parent of any bundle that doesn't specify a dependency.

Reply:
It is needed if any component has a Custom UI. Otherwise, the NAR plugin is unable to generate the documentation needed. While NiFi will create the linkage for you, it can do so because it knows that the NAR is in the "lib" directory. Without this, the plugin doesn't know what version of the jetty bundle to use. This will be necessary for any component that contains a Custom UI. It is, however, considered a best practice already to include this in such a case, because you are explicitly depending on those classes being in the classpath, so you should explicitly declare it.